.github/workflows/build-gateway.yml

```diff
@@ -0,0 +1,98 @@
+
+on:
+  workflow_dispatch:
+
+  repository_dispatch:
+    types:
+      - webhook
+
+  push:
+    paths:
+      - "apps/**/graph/*.graphqls"
+      - "apps/**/graph/struct-to-graphql/*.graphqls"
+      - "apps/gateway"
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  docker-builds:
+    runs-on: ubuntu-latest
+    name: Deploy to Docker Image
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Install Task
+        uses: arduino/setup-task@v1
+        with:
+          version: 3.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Rover
+        run: |
+          curl -sSL https://rover.apollo.dev/nix/v0.23.0-rc.3 | sh
+          
+          # Add Rover to the $GITHUB_PATH so it can be used in another step
+          # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path
+          echo "$HOME/.rover/bin" >> $GITHUB_PATH
+
+      - name: Generate Schema
+        run: |
+          cd apps/gateway
+          task generate-schema
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create Image Tag from branch name
+        if: startsWith(github.ref, 'refs/heads/release')
+        run: |
+          set +e
+          IMAGE_TAG=$(echo ${GITHUB_REF#refs/heads/} | sed 's/release-//g')
+          echo "$IMAGE_TAG" | grep -i '\-nightly$'
+          if [ $? -ne 0 ]; then
+            IMAGE_TAG="$IMAGE_TAG-nightly"
+          fi
+          set -e
+
+          echo "IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV
+          echo "OVERRIDE_PUSHED_IMAGE=true" >> $GITHUB_ENV
+
+      - name: Create Image Tag from tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          IMAGE_TAG=$(echo ${GITHUB_REF#refs/tags/})
+
+          echo "IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV
+          echo "OVERRIDE_PUSHED_IMAGE=false" >> $GITHUB_ENV
+
+      - name: Build & Push Image
+        if: startsWith(github.ref, 'refs/heads/release') || startsWith(github.ref, 'refs/tags/')
+        run: |
+          set +e
+          image_name="ghcr.io/${{ github.repository }}/gateway"
+
+          docker manifest inspect $image_name:$IMAGE_TAG
+          exit_status=$?
+          if [ $exit_status -eq 0 ]; then
+            [ "$OVERRIDE_PUSHED_IMAGE" = "false" ] && echo "image ($image_name:$IMAGE_TAG) already exists, and override image is disable, exiting" && exit 0
+            echo "image exists, but override pushed image is set to true. proceeding with building image"
+          fi
+
+          set -e
+          
+          cd apps/gateway
+          task container:build-and-push Image=$image_name:$IMAGE_TAG
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Validate the IMAGE_TAG format for Docker builds.

Ensure the IMAGE_TAG generated conforms to Docker's image tag naming conventions, especially since it's derived from branch names or tags which might not always adhere to these conventions.

```suggestion
          sanitized_image_tag=$(echo "$IMAGE_TAG" | tr '/' '-' | tr '[:upper:]' '[:lower:]')
          task container:build-and-push Image=$image_name:$sanitized_image_tag
```